main.go

```diff
@@ -160,10 +183,10 @@ func main() {
 	prometheus.MustRegister(version.NewCollector("snmp_exporter"))
 
 	// Bail early if the config is bad.
-	var err error
-	sc.C, err = config.LoadFile(*configFile)
+	err := sc.ReloadConfig(*configFile)
 	if err != nil {
 		level.Error(logger).Log("msg", "Error parsing config file", "err", err)
+		level.Error(logger).Log("msg", "Possible old config file, see https://github.com/prometheus/snmp_exporter/blob/main/auth-split-migration.md")
```

Comment:
This should be more explicit and a clear yes/no?

Reply:
Agreed, it's not an easy upgrade. But the end result is worth it and more future-proof.